# Topic 164277 (ID: 164277)

**Shashannk** (_2025-02-15T16:18:02.598Z_)

[@Jivraj](/u/jivraj) is it fine if i have my AIPROXY_TOKEN in my code instead of getting it as environment variable?

---

**Debjeet Singha** (_2025-02-15T16:20:06.323Z_)

if you do that then during evaluation, it would use your credit limit. if it gets exhausted, you may face problems. [@23f2003413](/u/23f2003413)

---

**Tushar Jalan ** (_2025-02-15T16:21:10.108Z_)

[image266×559 12.5 KB](https://europe1.discourse-cdn.com/flex013/uploads/iitm/original/3X/c/e/ce7f8f838b86960153991fdea76f15b4a50f80f7.png "image")

  
this is what i am doing first using the podman given in the portal and then running the evaluate.py file

---

**Shashannk** (_2025-02-15T16:21:31.012Z_)

ahhh okay, but i am getting an error while trying to fetch the token as an environment variable. any suggestions to fix this issue?

---

**Debjeet Singha** (_2025-02-15T16:22:17.623Z_)

you can use python-dotenv. check that out.

---

**Shashannk** (_2025-02-15T16:23:07.849Z_)

tried that still not getting T_T anyways thanks mate!

---

**Jivraj Singh Shekhawat** (_2025-02-15T16:25:07.138Z_)

No don’t do that, just follow the procedure.  
Two problems with keeping token in file.

  1. It will become public after you push to github.
  2. While running evaluation script after submission your token might run out of credits.

---

**Divjot Singh** (_2025-02-15T16:27:24.797Z_)

ohh yes, didn’t think it through xD

---

**Anulekha Pandi S** (_2025-02-15T16:29:14.206Z_)

I am facing the same error. and I have checked for solutions and found none. Did you resolve it? If yes can you please guide me through it?

---

**Shashannk** (_2025-02-15T16:34:00.432Z_)

{  
“detail”: “Error code: 401 - {‘error’: {‘message’: ‘Your authentication token is not from a valid issuer.’, ‘type’: ‘invalid_request_error’, ‘param’: None, ‘code’: ‘invalid_issuer’}}”  
} getting this error sir

---

**Andrew David** (_2025-02-15T16:40:47.374Z_)

[github.com](https://github.com/ANdIeCOOl/TDS-Project-1/tree/main/tds-project-1/app)

### [TDS-Project-1/tds-project-1/app at main · ANdIeCOOl/TDS-Project-1](https://github.com/ANdIeCOOl/TDS-Project-1/tree/main/tds-project-1/app)

Contribute to ANdIeCOOl/TDS-Project-1 development by creating an account on GitHub.

i keep updating, check this

---

**Ayush Tiwari** (_2025-02-15T16:47:32.801Z_)

Please extend deadline by 1 day. i just got discharged from hospital today, was suffering from liver problem since some days and got high heart beat due to a medicine unrelated to liver and made me got admitted@Jivraj

---

**Andrew David** (_2025-02-15T16:49:06.628Z_)

11:59 + 5 hours atthe most, [@s.anand](/u/s.anand) ?

---

**Jayakrishnan** (_2025-02-15T20:11:45.990Z_)

11 posts were split to a new topic: [Project 1 - Casual banter](/t/project-1-casual-banter/167344)

---

**shivam dubey** (_2025-02-15T16:59:47.955Z_)

[@Jivraj](/u/jivraj) sir [@carlton](/u/carlton) sir do have to add datagen in the docker container?  
As when I’m running it locally, it gives 9/10, but when I pull it from Hub and run eval, it says:detail": “[Errno 2] No such file or directory: ‘/data/datagen.py’”

---

**Shashannk** (_2025-02-15T17:03:12.600Z_)

[image706×193 6.15 KB](https://europe1.discourse-cdn.com/flex013/uploads/iitm/original/3X/b/9/b9a4995efdbe57c4d2d865982896333f8faf0c8c.png "image")

  
someone please help me fix this error

---

**Rohit Garg** (_2025-02-15T17:10:19.057Z_)

[@carlton](/u/carlton) can you pl merge this

[github.com/sanand0/tools-in-data-science-public](https://github.com/sanand0/tools-in-data-science-public/pull/7)

####  [Update evaluate.py with correct link of datagen.py for task `A1`](https://github.com/sanand0/tools-in-data-science-public/pull/7)

`tds-2025-01` ← `rohitxiitm:patch-1`

opened 10:56AM - 15 Feb 25 UTC

[ rohitxiitm ](https://github.com/rohitxiitm)

[ +1 -1 ](https://github.com/sanand0/tools-in-data-science-public/pull/7/files)

ppl are facing issues in evaluate.py for task A2

---

**Rohit Garg** (_2025-02-15T17:15:51.605Z_)

folks, need a confirmation. i don’t know but i heard it from someone or somewhere.  
we cannot send json in response, if it is success ? need to send text

is that really the case ?

---

**Anand S** (_2025-02-15T17:21:01.750Z_)

[@rohitgarg](/u/rohitgarg) \- thanks for this. Merged your PR pointing to the correct link for `evaluate.py`

---

**23F3004407 RATANPRIYA SINGH** (_2025-02-15T18:07:41.953Z_)

Sir from which session to which session is about tds project?

---

**Andrew David** (_2025-01-31T11:10:36.199Z_)

Please give the date, time and agenda also please.

---

**Carlton D'Silva** (_2025-01-31T11:38:24.014Z_)

Yes sir ,

As soon as we know we will send an announcement.

Kind regards.

---

**Andrew David** (_2025-02-01T06:48:06.736Z_)

the model keeps wrong answer, it says uvicorn for uv and has no info on how to run uv even after explicitly giving instructions(basically an older model) , basic “ls” command is also wrong, among other things. You can check your logs with respect to my api key.  
Do you think we could access a better model?

Maybe Download Deepseek 70b or even 671b and create an api while y’all run the model locally, in the long it would be cheaper for the course?  
because the model doesn’t know basic commands after telling how to do it.  
So if the model gives us wrong commands 2/3 times then how would we even solve the question.  
I spent a week on this just saying  
[@s.anand](/u/s.anand) [@carlton](/u/carlton) [@Jivraj](/u/jivraj)

---

**Andrew David** (_2025-02-01T07:03:38.242Z_)

sent pull request maybe accept it then please

---

**Andrew David** (_2025-02-01T07:50:52.938Z_)

[ ](https://www.youtube.com/watch?v=sdg4N-H4BR0)

can we have the code for this session please?  
[@Jivraj](/u/jivraj) [@carlton](/u/carlton)

---

**Andrew David** (_2025-02-02T08:46:20.663Z_)

i need some help can you send me your repo?

---

**Vivek Rekha Ashoka** (_2025-02-02T19:19:05.604Z_)

Hello, I recently started working on the project. I understood how to do all the phase A tasks on a high level but I’m struggling to start the implementation of the first task in phase A. I’m confused mainly about how the /data directory is supposed to be created, I don’t know how to generate the data and a little confused about the output formats. I would appreciate if I could get in contact with anyone who could guide me in the right direction.

---

**Hriday Pradhan** (_2025-02-03T06:42:50.121Z_)

Hello everyone, [@s.anand](/u/s.anand) [@carlton](/u/carlton)  
I had a few queries regarding the project;

  1. I am preloading my docker image with uv and generating the /data files when the container is ran. For task A1, I am automating my server to remove the /data directory that’s already present and run datagen.py again. Is this fine?
  2. For /read endpoint, is there a standard for parameters like “path=/data/format.md” or the parameter could be a plain english sentence like “path=show the data in format.md”?
  3. Are we concerned about what’s shown on the console if I run a /run command as long as it gets the job done?
  4. For tasks A1-10, are the file paths provided in the project doc standard or even they’re flexible? Ex. “Count the number of Wednesdays in file /data/format.md, and write just the number to /data/out.txt”

---

**Andrew David** (_2025-02-03T08:00:58.755Z_)

+1

---

**24DS1000121 ULAGAOOZHIAN** (_2025-02-03T08:54:03.151Z_)

Dear Sir,  
Can we have a mentorship program for TDS for those who have no experience in programming like me ?  
thanks & regards.  
ULAGAOOZHIAN

---

**Dewang Gandhi** (_2025-02-02T10:36:40.360Z_)

For Project-1 to complete, it requires:  
"You MUST complete ALL these 3 steps to get a score. Failure to do so will result in getting 0 in the project. If you do not do ALL these 3 steps before the deadline, there will be no appeal available.  
• Fill the form that is on the Project Page  
But I did not get the form; where is it? While I checked inside the project pages also.

---

**Carlton D'Silva** (_2025-02-03T13:02:45.150Z_)

Hi Dewang,

[Screenshot 2025-02-03 at 6.27.39 pm 12268×1766 491 KB](https://europe1.discourse-cdn.com/flex013/uploads/iitm/original/3X/9/2/9286f3dcf5984d514cf6a40996bd5040f5d9c306.png "Screenshot 2025-02-03 at 6.27.39 pm 1")

Please _read_ the Project page Deliverables carefully as well as the Evaluation Pre - Requisites.

Kind regards

---

**Andrew David** (_2025-02-04T09:04:40.260Z_)

[github.com/ANdIeCOOl/TDS-Project1-Ollama_FastAPI-](https://github.com/ANdIeCOOl/TDS-Project1-Ollama_FastAPI-/blob/main/README.md)

#### [README.md](https://github.com/ANdIeCOOl/TDS-Project1-Ollama_FastAPI-/blob/main/README.md)

[`main`](https://github.com/ANdIeCOOl/TDS-Project1-Ollama_FastAPI-/blob/main/README.md)
    
    
    # TDS-Project1-Ollama_FastAPI-
    ## Info
    - Create codespaces on main or evalution script branch
    Use history.txt to get sqlite to version 3.45.3 into bash session 
       - 64  export PATH=/opt/conda/bin:$PATH
       - 65  export LD_LIBRARY_PATH=/opt/conda/lib:$LD_LIBRARY_PATH
       - 66  python3 -c "import sqlite3; print(sqlite3.sqlite_version)"
    
    - cd to latest_ai_development and run cmd [ crewai run] which set up server 
    - Then in a separate bash terminal run "python evaluate.py" 
    - also make sure to enter openai or sanand api key in crew.py
    
    # Simple history of commands
    1. Terminal 1 
        - 1  python3 -c "import sqlite3; print(sqlite3.sqlite_version)"
        - 2  export PATH=/opt/conda/bin:$PATH
        - 3  export LD_LIBRARY_PATH=/opt/conda/lib:$LD_LIBRARY_PATH
        - 4  python3 -c "import sqlite3; print(sqlite3.sqlite_version)"
        - 5  cd latest_ai_development/
        - 7  pip install crewai crewai-tools
    

This file has been truncated. [show original](https://github.com/ANdIeCOOl/TDS-Project1-Ollama_FastAPI-/blob/main/README.md)

My take on autonomous agents. Limited by model capabilities to some extent. Will use function calling hence forth but here is a quick look at using crewai for agent tasks.

---

**Guddu Kumar Mishra ** (_2025-02-04T09:55:02.489Z_)

Sir [@carlton](/u/carlton) [@Jivraj](/u/jivraj) just saying,  
If possible Please do 40-50% of project in upcoming live sessions so that we all have atleast something to submit.

---

**Arjun G** (_2025-02-05T16:32:16.103Z_)

I am using ubuntu. How do I use python 3.13. It says my python version is 3.12 even after installing python 3.13  
Someone please help

---

**Shivaditya Bhattacharya** (_2025-02-05T18:38:02.561Z_)

[@s.anand](/u/s.anand) sir, I see that the project 1 timeline was changed from February 7 - 17, 2025 to January 17 - February 15 which undoubtedly is a good increase in duration. However, I have my GATE DA exam on Feb 15 and the exam center is unexpectedly far. So, I request you to consider pushing the deadline to at least Feb 16. If not, I’ll still do my best.

---

**Hriday Pradhan** (_2025-02-06T07:04:26.179Z_)

Hello! [@carlton](/u/carlton) [@s.anand](/u/s.anand)

Is the proxy server down right now?  
I am getting this error when I am accessing the endpoint:

{‘id’: ‘chatcmpl-Axq55TzulOVjHYuXYIhkRQzCC3PNl’, ‘object’: ‘chat.completion’, ‘created’: 1738824915, ‘model’: ‘gpt-4o-mini-2024-07-18’, ‘choices’: [{‘index’: 0, ‘message’: {‘role’: ‘assistant’, ‘content’: …, ‘costError’: ‘crypto.createHash is not a function’}

Or, do I have to install crypto module?

---

**Anand S** (_2025-02-06T07:29:30.529Z_)

[@21f3002390](/u/21f3002390) \- AI Proxy is working and you _did_ get the result. You can ignore any `costError`. It won’t happen in the future anyway.

**What’s happening?** I was trying to generate a unique hash for each request, as a precursor to caching requests. But I made a mistake in the code. Specifically, `crypto.createHash` is not supported in CloudFlare. [I fixed that](https://github.com/sanand0/aiproxy/commit/5943b6d355deffff88ac07d17aa0c6969cacc3d5) by removing this. I’ll introduce caching later if required.

---

**Sarang Tambe** (_2025-02-06T09:28:32.920Z_)

For the question #A8 on recognizing the credit card number in the image, Open AI doesn’t seem to be recognizing the number correctly and as a result the evaluation is failing. What should be the solution?  


[image913×498 13.6 KB](https://europe1.discourse-cdn.com/flex013/uploads/iitm/original/3X/e/a/eab0a8c362c564a00917bb033bce6ad5ba40d103.png "image")

---

**DEEPANSHU** (_2025-02-06T12:31:43.426Z_)

When will live sessions for demo project start? If started please provide link for that as I am unable to get what the project is about and what are the initial steps to start project.

---

**Divyasree** (_2025-01-27T14:02:29.072Z_)

by clicking the link

[image1198×136 9.49 KB](https://europe1.discourse-cdn.com/flex013/uploads/iitm/original/3X/3/2/32bd53681054ab17de6350c49f68b405acd538b9.png "image")

[image1750×581 70.9 KB](https://europe1.discourse-cdn.com/flex013/uploads/iitm/original/3X/9/3/937562cc32dc76a582f6845678b048730622d388.png "image")

  
I am getting this opened.

---

**Jivraj Singh Shekhawat** (_2025-01-27T21:30:57.919Z_)

Hi [@Divya1](/u/divya1) ,

There won’t be any project1 page such as GA1s, there is a google form(which can be found in same page) which needs to be filled after you do project1.

---

**Jivraj Singh Shekhawat** (_2025-01-27T21:57:49.221Z_)

Hi [@23f2005325](/u/23f2005325) ,

Extracting details from credit cards is sensitive, try using strong prompts or take code from LLM and execute it in script.

kind regards

---

**Andrew David** (_2025-01-28T08:28:17.260Z_)

Regarding Wednenday 9-10 pm live session, maybe the instructors could also discuss how to use docker as a virtual environment using maybe ollama(local llm as now there is deepseek opensource, i doubt we would need to use openai for testing, just for production(test submission) would be enough) and also some agent(langchain, autogen, crewai) just a quick how-to on setting up and problems while setting up if possible

More resources on docker. Using docker as a virtual environment. Editing and executing code in Dockerfiles (like when you change code in src a web framework automatically reloads page(hot reload)), something along the lines of this .

[@carlton](/u/carlton) [@Jivraj](/u/jivraj)

---

**Jivraj Singh Shekhawat** (_2025-01-28T11:55:55.799Z_)

23f1002382:

> Regarding Wednenday 9-10 pm live session, maybe the instructors could also discuss how to use docker as a virtual environmen

In Tuesday’s(21 January) session we had discussed docker towards ending of session.  
What was discussed in that live session regarding docker:

  1. Search for existing containers on repositories such as dockerhub.
  2. Pull an existing docker image.
  3. Run that image inside a container.
  4. Enter to that container and modify something(such as installing python inside a ubuntu container, for customization or create some file)
  5. Once done you can commit it.
  6. And push customized container’s image to docker hub.



Regarding local models running for project1, it’s a good idea, we will see if it’s possible to discuss in session.

---

**Divyasree** (_2025-01-28T18:07:19.143Z_)

In the google forms , I have 2 questions in one form now to submit should it is compulsory that to answer the both the questions?

---

**Carlton D'Silva** (_2025-01-29T02:57:18.959Z_)

Hi [@Divya1](/u/divya1)

[Screenshot 2025-01-29 at 8.19.05 am1738×982 122 KB](https://europe1.discourse-cdn.com/flex013/uploads/iitm/original/3X/3/5/35e7ce763c7605e99ee1fad3906e1cd31d094b31.png "Screenshot 2025-01-29 at 8.19.05 am")

Please do very carefully all things mentioned in the Deliverables as well as look at the Evaluation Section.  


[Screenshot 2025-01-29 at 8.26.08 am1460×496 45.5 KB](https://europe1.discourse-cdn.com/flex013/uploads/iitm/original/3X/8/9/898ab28ebe773e40fb3ad9b98c71ce4c5d063c09.png "Screenshot 2025-01-29 at 8.26.08 am")

We had a session on 28th Jan introducing all the important aspects of Project.

If you do not do everything exactly as mentioned **especially the pre - requisites** mentioned in the Evaluation section you will get 0 in the project and _there will be no appeal_ for failing to meet the pre - requisites of the evaluation criteria.

In order for us to evaluate the project you have to provide the deliverables mentioned above.

Kind regards

---

**Andrew David** (_2025-01-29T06:32:45.816Z_)

* * *

**Subject:** Request to Add Instructors to Private GitHub Repo

**Message:**  
_"Dear [Instructors’ Names],_

_I’ve set up the environment and dependencies for the project and was wondering if it would be appropriate to add you to my private GitHub repository. I’d appreciate any guidance on improving performance, scalability, and design principles. Please let me know if this is feasible or if there’s a more suitable way to seek feedback. Apologies if this request is out of scope._

_Thank you for your time!_

_Best,_  
[Your Name]"*

ChatGPT can make mistakes. Check important info.

---

**Anand S** (_2025-01-29T10:41:16.527Z_)

[@23f1002382](/u/23f1002382) \- You’re welcome to use the evaluation script in this post for private repos.

[Project 1 - LLM-based Automation Agent - Discussion Thread [TDS Jan 2025]](https://discourse.onlinedegree.iitm.ac.in/t/project-1-llm-based-automation-agent-discussion-thread-tds-jan-2025/164277/21) [Tools in Data Science](/c/courses/tds-kb/34)

> A sample evaluation script for Project 1 tasks A1-A10 is available at [tools-in-data-science-public/project-1 at tds-2025-01-project-1-wip · sanand0/tools-in-data-science-public · GitHub](https://github.com/sanand0/tools-in-data-science-public/tree/tds-2025-01-project-1-wip/project-1) You can use this to validate your code for Project 1. Please note: This is a sample. It WILL change. Don’t rely on the dataset being the same. It WILL change. LLMs give different results each time they are called. Make sure: Your code gives correct results reliably (i.e. try a few times) Change the task in t… 

For public repos submitted in the form, I’ll run this script over the weekend and share preliminary results.

---

**Andrew David** (_2025-01-29T11:29:04.323Z_)

T h a n k y o u sir.

---

**Joel Jeffrey** (_2025-01-30T06:20:34.650Z_)

For A6, /data/docs/ has subfolders with .md files from which we have to extract the heading level 1’s (#) right? Apparently there are few files with different content but the same name. Can someone confirm the same? If yes how to address these files [@Jivraj](/u/jivraj) [@carlton](/u/carlton)

---

**Andrew David** (_2025-01-30T06:26:20.605Z_)

I had set up the environment and dependencies and everything was working fine. When i tried to recreate it from scratch in a new codespace it broke. I fixed almost everything except this error
    
    
    @ANdIeCOOl ➜ /workspaces/TDS-Project-1 (main) $ crewai create crew b2b
    Traceback (most recent call last):
      File "/home/codespace/.python/current/bin/crewai", line 5, in <module>
        from crewai.cli.cli import crewai
      File "/usr/local/python/3.12.1/lib/python3.12/site-packages/crewai/__init__.py", line 3, in <module>
        from crewai.agent import Agent
      File "/usr/local/python/3.12.1/lib/python3.12/site-packages/crewai/agent.py", line 7, in <module>
        from crewai.agents import CacheHandler
      File "/usr/local/python/3.12.1/lib/python3.12/site-packages/crewai/agents/__init__.py", line 2, in <module>
        from .parser import CrewAgentParser
      File "/usr/local/python/3.12.1/lib/python3.12/site-packages/crewai/agents/parser.py", line 6, in <module>
        from crewai.utilities import I18N
      File "/usr/local/python/3.12.1/lib/python3.12/site-packages/crewai/utilities/__init__.py", line 13, in <module>
        from .embedding_configurator import EmbeddingConfigurator
      File "/usr/local/python/3.12.1/lib/python3.12/site-packages/crewai/utilities/embedding_configurator.py", line 4, in <module>
        from chromadb import Documents, EmbeddingFunction, Embeddings
      File "/usr/local/python/3.12.1/lib/python3.12/site-packages/chromadb/__init__.py", line 6, in <module>
        from chromadb.auth.token_authn import TokenTransportHeader
      File "/usr/local/python/3.12.1/lib/python3.12/site-packages/chromadb/auth/token_authn/__init__.py", line 24, in <module>
        from chromadb.telemetry.opentelemetry import (
      File "/usr/local/python/3.12.1/lib/python3.12/site-packages/chromadb/telemetry/opentelemetry/__init__.py", line 13, in <module>
        from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import OTLPSpanExporter
      File "/usr/local/python/3.12.1/lib/python3.12/site-packages/opentelemetry/exporter/otlp/proto/grpc/trace_exporter/__init__.py", line 25, in <module>
        from opentelemetry.exporter.otlp.proto.grpc.exporter import (  # noqa: F401
      File "/usr/local/python/3.12.1/lib/python3.12/site-packages/opentelemetry/exporter/otlp/proto/grpc/exporter.py", line 72, in <module>
        from opentelemetry.sdk.metrics.export import MetricsData
      File "/usr/local/python/3.12.1/lib/python3.12/site-packages/opentelemetry/sdk/metrics/__init__.py", line 16, in <module>
        from opentelemetry.sdk.metrics._internal import Meter, MeterProvider
      File "/usr/local/python/3.12.1/lib/python3.12/site-packages/opentelemetry/sdk/metrics/_internal/__init__.py", line 56, in <module>
        from opentelemetry.sdk.metrics._internal.measurement_consumer import (
      File "/usr/local/python/3.12.1/lib/python3.12/site-packages/opentelemetry/sdk/metrics/_internal/measurement_consumer.py", line 29, in <module>
        from opentelemetry.sdk.metrics._internal.metric_reader_storage import (
      File "/usr/local/python/3.12.1/lib/python3.12/site-packages/opentelemetry/sdk/metrics/_internal/metric_reader_storage.py", line 26, in <module>
        from opentelemetry.sdk.metrics._internal._view_instrument_match import (
      File "/usr/local/python/3.12.1/lib/python3.12/site-packages/opentelemetry/sdk/metrics/_internal/_view_instrument_match.py", line 22, in <module>
        from opentelemetry.sdk.metrics._internal.aggregation import (
      File "/usr/local/python/3.12.1/lib/python3.12/site-packages/opentelemetry/sdk/metrics/_internal/aggregation.py", line 48, in <module>
        from opentelemetry.sdk.metrics._internal.exponential_histogram.mapping.exponent_mapping import (
      File "/usr/local/python/3.12.1/lib/python3.12/site-packages/opentelemetry/sdk/metrics/_internal/exponential_histogram/mapping/exponent_mapping.py", line 25, in <module>
        from opentelemetry.sdk.metrics._internal.exponential_histogram.mapping.ieee_754 import (
      File "/usr/local/python/3.12.1/lib/python3.12/site-packages/opentelemetry/sdk/metrics/_internal/exponential_histogram/mapping/ieee_754.py", line 15, in <module>
        from ctypes import c_double, c_uint64
      File "/usr/local/python/3.12.1/lib/python3.12/ctypes/__init__.py", line 8, in <module>
        from _ctypes import Union, Structure, Array
    ImportError: /usr/local/python/3.12.1/lib/python3.12/lib-dynload/_ctypes.cpython-312-x86_64-linux-gnu.so: undefined symbol: ffi_type_uint32, version LIBFFI_BASE_7.0
    

i updated the libffi package using sudo but while breaking something else can someone pls help me? [@carlton](/u/carlton) [@Jivraj](/u/jivraj) [@s.anand](/u/s.anand)  
  
  
  
  
  


history of commands in new codespace
    
    
        1  crewai --version
        2  pip install crewai crewai-tools
        3  python3 -c "import sqlite3; print(sqlite3.sqlite_version)"
        4  export PATH=/opt/conda/bin:$PATH
        5  export LD_LIBRARY_PATH=/opt/conda/lib:$LD_LIBRARY_PATH
        6  python3 -c "import sqlite3; print(sqlite3.sqlite_version)"
        7  crewai create crew <project_name>
        8  crewai create crew b2b
        9  history
    

  
  


UPDATE: IT’s WORKING if you do this in order
    
    
        1  python3 -c "import sqlite3; print(sqlite3.sqlite_version)"
        2  export PATH=/opt/conda/bin:$PATH
        3  export LD_LIBRARY_PATH=/opt/conda/lib:$LD_LIBRARY_PATH
        4  python3 -c "import sqlite3; print(sqlite3.sqlite_version)"
        5  pip install --no-cache-dir --force-reinstall typing_extensions pydantic crewai crewai-tools
        6  conda install -c conda-forge typing_extensions
        7  exec bash
        8  crewai create crew "Project 1 - LLM-based Automation Agent"
    

Something about different environment conda and python can the instructors please help me understand it(resources ), so i can trouble shoot this later with better accuracy come precision

---

**Andrew David** (_2025-01-30T12:51:19.538Z_)

evaluate.py  
TDS course repo

[github.com](https://github.com/sanand0/tools-in-data-science-public/tree/tds-2025-01-project-1-wip/project-1)

### [tools-in-data-science-public/project-1 at tds-2025-01-project-1-wip ·...](https://github.com/sanand0/tools-in-data-science-public/tree/tds-2025-01-project-1-wip/project-1)

Contribute to sanand0/tools-in-data-science-public development by creating an account on GitHub.

line 20
    
    
    from datagen import (
        get_markdown,
        get_dates,
        get_contacts,
        get_logs,
        get_docs,
        get_email,
        get_credit_card,
        get_comments,
        get_tickets,
    )
    

but we get datagen.py only in a1 task  
line 69
    
    
    async def a1(email: str, **kwargs):
        await run(
            f"""
    Install `uv` (if required) and run the script `https://raw.githubusercontent.com/sanand0/tools-in-data-science-public/tds-2025-01/datagen.py`
    with `{email}` as the only argument
    """
        )
        return email in await read("/data/format.md")
    

The issue is **importing`datagen` before ensuring it exists**

just checking

[@carlton](/u/carlton) [@Jivraj](/u/jivraj)

---

**Jivraj Singh Shekhawat** (_2025-01-30T21:37:48.839Z_)

Hi [@23f1002382](/u/23f1002382),

Yes datagen.py must be present in same directory from where you are executing evaluate.py.

Oh, You trying to use crewai locally for Project1  
kind regards

---

**Jivraj Singh Shekhawat** (_2025-01-30T21:56:52.077Z_)

Hi [@JoelJeffrey](/u/joeljeffrey) ,

Filepath is unique for every file, which needs to be inserted to json file.

---

**Joel Jeffrey** (_2025-01-31T06:55:55.139Z_)

Ok. So just to confirm, since there are files with the same name, the json file should map the filepath and not the filename to the title right?  


[Screenshot from 2025-01-31 12-25-29790×117 19.9 KB](https://europe1.discourse-cdn.com/flex013/uploads/iitm/original/3X/d/3/d3ebea3238860bad920a47ff55ac33cb02ad2d63.png "Screenshot from 2025-01-31 12-25-29")

---

**Andrew David** (_2025-01-31T08:40:50.303Z_)

no crewai, it takes really long i put time out for 300 secs(in run(task:str) in evaluate.py) still sometimes its not enough. I’ll try with autogen next and then langchain

---

**Guddu Kumar Mishra ** (_2025-01-31T08:41:45.108Z_)

INFO:     127.0.0.1:65085 - "GET /read?path=/data/format.md HTTP/1.1" 200 OK
    data/format.md 81ms
    INFO:     127.0.0.1:65149 - "POST /run?task=%0AFormat+the+contents+of+%60%2Fdata%2Fformat.md%60+using+%60prettier%403.4.2%60%2C+updating+the+file+in-place%0A HTTP/1.1" 200 OK
    INFO:     127.0.0.1:65251 - "GET /read?path=/data/format.md HTTP/1.1" 200 OK
    INFO:     127.0.0.1:65263 - "POST /run?task=The+file+%60%2Fdata%2Fdates.txt%60+contains+a+list+of+dates%2C+one+per+line.+Count+the+number+of+Wednesdays+in+the+list%2C+and+write+just+the+number+to+%60%2Fdata%2Fdates-wednesdays.txt%60 HTTP/1.1" 200 OK
    INFO:     127.0.0.1:65298 - "GET /read?path=/data/dates-wednesdays.txt HTTP/1.1" 200 OK
    INFO:     127.0.0.1:65312 - "POST /run?task=Sort+the+array+of+contacts+in+%60%2Fdata%2Fcontacts.json%60+by+%60last_name%60%2C+then+%60first_name%60%2C+and+write+the+result+to+%60%2Fdata%2Fcontacts-sorted.json%60 HTTP/1.1" 200 OK
    INFO:     127.0.0.1:65350 - "GET /read?path=/data/contacts-sorted.json HTTP/1.1" 200 OK
    INFO:     127.0.0.1:65361 - "POST /run?task=Write+the+first+line+of+the+10+most+recent+%60.log%60+file+in+%60%2Fdata%2Flogs%2F%60+to+%60%2Fdata%2Flogs-recent.txt%60%2C+most+recent+first HTTP/1.1" 200 OK
    INFO:     127.0.0.1:65390 - "GET /read?path=/data/logs-recent.txt HTTP/1.1" 200 OK
    INFO:     127.0.0.1:65402 - "POST /run?task=Find+all+Markdown+%28%60.md%60%29+files+in+%60%2Fdata%2Fdocs%2F%60.%0AFor+each+file%2C+extract+the+first+occurrance+of+each+H1+%28i.e.+a+line+starting+with+%60%23+%60%29.%0ACreate+an+index+file+%60%2Fdata%2Fdocs%2Findex.json%60+that+maps+each+filename+%28without+the+%60%2Fdata%2Fdocs%2F%60+prefix%29+to+its+title%0A%28e.g.+%60%7B%22README.md%22%3A+%22Home%22%2C+%22path%2Fto%2Flarge-language-models.md%22%3A+%22Large+Language+Models%22%2C+...%7D%60%29 HTTP/1.1" 200 OK
    INFO:     127.0.0.1:65436 - "GET /read?path=/data/docs/index.json HTTP/1.1" 200 OK
    INFO:     127.0.0.1:65452 - "POST /run?task=%60%2Fdata%2Fcredit_card.png%60+contains+a+credit+card+number.+Pass+the+image+to+an+LLM%2C+have+it+extract+the+card+number%2C+and+write+it+without+spaces+to+%60%2Fdata%2Fcredit-card.txt%60 HTTP/1.1" 500 Internal Server Error
    INFO:     127.0.0.1:65482 - "GET /read?path=/data/credit-card.txt HTTP/1.1" 500 Internal Server Error
    INFO:     127.0.0.1:65503 - "POST /run?task=The+SQLite+database+file+%60%2Fdata%2Fticket-sales.db%60+has+a+%60tickets%60+with+columns+%60type%60%2C+%60units%60%2C+and+%60price%60.+Each+row+is+a+customer+bid+for+a+concert+ticket.+What+is+the+total+sales+of+all+the+items+in+the+%22Gold%22+ticket+type%3F+Write+the+number+in+%60%2Fdata%2Fticket-sales-gold.txt%60 HTTP/1.1" 200 OK
    INFO:     127.0.0.1:49154 - "GET /read?path=/data/ticket-sales-gold.txt HTTP/1.1" 200 OK
    

result after running evaluate.py:

Score: 0 / 10

why sir [@Jivraj](/u/jivraj) [@Saransh_Saini](/u/saransh_saini) what is the problem here??  
please do a live session of complete project process with one or two tasks if possible

---

**Carlton D'Silva** (_2025-01-31T09:04:35.954Z_)

Hi Guddu,

We are planning several project sessions in order to show the workflow of creating a successful project.

Although you are returning a 200 ok, the get request file must match the expectation. In other words after running the first task for example, has the new format.md been formatted correctly and matches the expected output.

In this case you would write out the the `expected` variable in the `evaluate.py` and see if `result` variable matches the `expected`. Then you can figure out what went wrong.

Kind regards

---

**Guddu Kumar Mishra ** (_2025-01-31T09:32:51.688Z_)

Ok sir  
But please try to take those sessions sooner  
Because it’s taking too much time for me to do any problem(plus two more courses and one oppe you know) .so I just want to build the project before deadline.

---
